### ISSUE TITLE
Compile messages to JSON

### ISSUE THREAD

**Issue by simlmx** [lingui/js-lingui#1866]
**Is your feature request related to a problem? Please describe.**
I'd like to compile the message files to JSON, instead of javascript (or typescript). This way I could load the compiled messages without needing to interpret javascript code.

**Describe proposed solution**
Although not documented, it seems like this is already almost fully implemented.
`"json"` is properly treated as a `compileNamespace` [here][1]. All that is missing is to properly set the file extension [here][2], I think. Currently it uses the `.js` fallback.

[1]: https://github.com/lingui/js-lingui/blob/107bd02edc26d6bb631923288db88d9036604cc8/packages/cli/src/api/compile.ts#L45
[2]: https://github.com/lingui/js-lingui/blob/107bd02edc26d6bb631923288db88d9036604cc8/packages/cli/src/api/catalog.ts#L209

I'm happy to contribute a PR if you like.

**Comment by thekip** [comment]
You can contribute, we will happily accept PRs

**Comment by vonovak** [comment]
> This way I could load the compiled messages without needing to interpret javascript code.

would you mind expanding on why you want this?

Please note that even the JS files are currently just `export const messages = JSON.parse(...)`

Thank you :)

**Comment by simlmx** [comment]
@vonovak 
> would you mind expanding on why you want this?

I have a pretty complex setup where other packages (other than the one in which the translations are compiled) need to consume the compiled message files, and it made my life easier if they are simply in JSON.

But since then I managed to make it work with compiled .mjs files!

> Please note that even the JS files are currently just export const messages = JSON.parse(...)

Yes I saw that! Any reason to not use the JSON files by default and save a couple of characters? It would make the compiled files easy to load anywhere (for instance in a python backend or I don't know).
